From 98a966a97290d18b06d48bc7d198f15c898de1cd Mon Sep 17 00:00:00 2001
From: Kent Overstreet <kent.overstreet@linux.dev>
Date: Sat, 14 Jan 2023 05:57:07 -0500
Subject: [PATCH 43/44] block/rq_qos: Use proper wait initializer

This is needed to avoid breakage with the new code tagging based latency
tracing.

Signed-off-by: Kent Overstreet <kent.overstreet@linux.dev>
---
 block/blk-rq-qos.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/block/blk-rq-qos.c b/block/blk-rq-qos.c
index 88f0fe7dc..396d5db91 100644
--- a/block/blk-rq-qos.c
+++ b/block/blk-rq-qos.c
@@ -245,10 +245,7 @@ void rq_qos_wait(struct rq_wait *rqw, void *private_data,
 		 cleanup_cb_t *cleanup_cb)
 {
 	struct rq_qos_wait_data data = {
-		.wq = {
-			.func	= rq_qos_wake_function,
-			.entry	= LIST_HEAD_INIT(data.wq.entry),
-		},
+		.wq = WAIT_FUNC_INITIALIZER(data.wq, rq_qos_wake_function),
 		.task = current,
 		.rqw = rqw,
 		.cb = acquire_inflight_cb,
-- 
2.39.0.rc2.1.gbd5df96b79

